File: ccif/src/java/other/org/himalay/ccif/Set_DSG_Mode1234.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:13 EST 2016")

public  class Set_DSG_Mode1234 extends   APDUBaseFactory.APDUBase { //Concrete type is Set_DSG_Mode1234


// member static classes
    public  static  class StatusfieldCheck extends BinStruct implements PublicBinMsg {




        // members variables
        // macAddress
        public MacAddress macAddress ;
        // remove_header_bytes
        public int remove_header_bytes ;

        // Parent of this object
        Set_DSG_Mode1234 parent;
        public StatusfieldCheck (Set_DSG_Mode1234 parent) // throws Exception
        {
            this.parent= parent;
            init();
        }
        public Set_DSG_Mode1234 getParent()
        {
            return parent;
        }

        private void init()
        {
            // Initialize macAddress
            macAddress = new MacAddress();
            // Initialize remove_header_bytes

        }

        public int readNoHeader(DataInputStream istream) throws IOException
        {

            return read(istream);
        }

        public int read(DataInputStream istream) throws IOException
        {
            preRead();
            int retVal= 0;

            // read macAddress
            retVal += macAddress.read(istream);
            // read remove_header_bytes
            {
                remove_header_bytes=istream.readUnsignedShort();
                retVal+=2;
            }

            postRead();
            return retVal;
        }


        public int write(DataOutputStream ostream) throws IOException
        {
            preWrite();
            int retVal= 0;

            {   /** fix dependent sizes for macAddress **/
            }


            // write macAddress
            if (macAddress!=null)retVal +=macAddress.write(ostream);
            // write remove_header_bytes
            ostream.writeShort(remove_header_bytes);
            retVal +=2;
            postWrite();
            return retVal;
        }

        public int dump(DumpContext dc) throws IOException
        {
            dc.indent();
            dc.getPs().print("StatusfieldCheck\n");
            dc.increaseIndent();
            int retVal= 0;
            // write macAddress
            if ( macAddress != null ) {
                dc.indent();
                dc.getPs().println("macAddress") ;
                retVal +=macAddress.dump(dc);
            }
            // write remove_header_bytes
            dc.indent();
            dc.getPs().println("remove_header_bytes="+remove_header_bytes+"(0x"+ Integer.toHexString(remove_header_bytes) + ")" );
            dc.decreaseIndent();
            return retVal;
        }


        // Getter for macAddress
        //public MacAddress getMacAddress()
        //{
        //    return macAddress ;
        //}


        // Setter for macAddress
        //public void setMacAddress(MacAddress val)
        //{
        //    this.macAddress= val;
        //}
        // Getter for remove_header_bytes
        //public int getRemove_header_bytes()
        //{
        //    return remove_header_bytes ;
        //}


        // Setter for remove_header_bytes
        //public void setRemove_header_bytes(int val)
        //{
        //    this.remove_header_bytes= val;
        //}


        public int getSize() throws IOException
        {
            DataOutputStream dos= new DataOutputStream(new NullStream());
            return this.write(dos);
        }



    }

// End of code


    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // operationalMode
    public short operationalMode ;
    // StatusfieldCheckInstance
    public StatusfieldCheck StatusfieldCheckInstance ;

    public Set_DSG_Mode1234 () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize operationalMode

        // Initialize StatusfieldCheckInstance
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read operationalMode
        {
            operationalMode=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read StatusfieldCheckInstance
        if ((operationalMode == 0x03) ||                (operationalMode ==0x04)) {
            StatusfieldCheckInstance = new StatusfieldCheck(this);
            retVal += StatusfieldCheckInstance.read(istream);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read operationalMode
        {
            operationalMode=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read StatusfieldCheckInstance
        if ((operationalMode == 0x03) ||                (operationalMode ==0x04)) {
            StatusfieldCheckInstance = new StatusfieldCheck(this);
            retVal += StatusfieldCheckInstance.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }


        {   /** fix dependent sizes for StatusfieldCheckInstance **/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write operationalMode
        ostream.writeByte(operationalMode);
        retVal +=1;
        // write StatusfieldCheckInstance
        if (StatusfieldCheckInstance!=null)retVal +=StatusfieldCheckInstance.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("Set_DSG_Mode1234\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write operationalMode
        dc.indent();
        dc.getPs().println("operationalMode="+operationalMode+"(0x"+ Integer.toHexString(operationalMode) + ")" );
        // write StatusfieldCheckInstance
        if ( StatusfieldCheckInstance != null ) {
            dc.indent();
            dc.getPs().println("StatusfieldCheckInstance") ;
            retVal +=StatusfieldCheckInstance.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for operationalMode
    //public short getOperationalMode()
    //{
    //    return operationalMode ;
    //}


    // Setter for operationalMode
    //public void setOperationalMode(short val)
    //{
    //    this.operationalMode= val;
    //}
    // Getter for StatusfieldCheckInstance
    //public StatusfieldCheck getStatusfieldCheckInstance()
    //{
    //    return StatusfieldCheckInstance ;
    //}


    // Setter for StatusfieldCheckInstance
    //public void setStatusfieldCheckInstance(StatusfieldCheck val)
    //{
    //    this.StatusfieldCheckInstance= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/Set_DSG_Mode1234.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:19 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "Set_DSG_Mode1234.h"

#ifdef org_himalay_ccif_Set_DSG_Mode1234__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;
// member static classes
// Constructor
Set_DSG_Mode1234::StatusfieldCheck::StatusfieldCheck() {
    references= 0;

    parent = NULL;

    // macAddress
    org_himalay_ccif_MacAddress__NEW( macAddress);
    // remove_header_bytes
    remove_header_bytes= (ui16)0;
}
// Destructor
Set_DSG_Mode1234::StatusfieldCheck::~StatusfieldCheck() {
    // macAddress
    org_himalay_ccif_MacAddress__DELETE( macAddress);
    // remove_header_bytes

    parent = NULL;
}

int Set_DSG_Mode1234::StatusfieldCheck::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // macAddress
    retVal += macAddress->read(istream);
    // remove_header_bytes
    {
        remove_header_bytes=istream.readUI16();
        retVal+=2;
    }
    return retVal;
}

int Set_DSG_Mode1234::StatusfieldCheck::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read macAddress
    retVal += macAddress->read(istream);
    // read remove_header_bytes
    {
        remove_header_bytes=istream.readUI16();
        retVal+=2;
    }

    return retVal;
}

int Set_DSG_Mode1234::StatusfieldCheck::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for macAddress **/
    }


    int retVal= 0;
    // write macAddress
    if ( macAddress != NULL ) {
        retVal +=macAddress->write(ostream);
    }
    // write remove_header_bytes
    ostream.writeUI16(remove_header_bytes);
    retVal +=2;

    return retVal;
}

int Set_DSG_Mode1234::StatusfieldCheck::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("StatusfieldCheck\n");
    dc.increaseIndent();
    int retVal= 0;
    // write macAddress
    if ( macAddress != NULL ) {
        dc.indent();
        dc.getPs().println("macAddress") ;
        retVal +=macAddress->dump(dc);
    }
    // write remove_header_bytes
    dc.indent();
    dc.getPs().print("remove_header_bytes=");
    dc.getPs().println((long)remove_header_bytes);
    dc.decreaseIndent();
    return retVal;
}


// Getter for macAddress
//org_himalay_ccif_MacAddress__PTR_TYPE Set_DSG_Mode1234::StatusfieldCheck::getMacAddress()
//{
//return this->macAddress;
//}

// Setter for macAddress
//void Set_DSG_Mode1234::StatusfieldCheck::setMacAddress(org_himalay_ccif_MacAddress__PTR_TYPE val)
//{
//this->macAddress= val;
//}

// Getter for remove_header_bytes
//ui16 Set_DSG_Mode1234::StatusfieldCheck::getRemove_header_bytes()
//{
//return this->remove_header_bytes;
//}

// Setter for remove_header_bytes
//void Set_DSG_Mode1234::StatusfieldCheck::setRemove_header_bytes(ui16 val)
//{
//this->remove_header_bytes= val;
//}

int Set_DSG_Mode1234::StatusfieldCheck::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->decreaseRef();
};;
#endif

// Constructor
Set_DSG_Mode1234::Set_DSG_Mode1234() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // operationalMode
    operationalMode= (ui8)0;
    // StatusfieldCheckInstance
    org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__INIT(StatusfieldCheckInstance)
}
// Destructor
Set_DSG_Mode1234::~Set_DSG_Mode1234() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // operationalMode

    // StatusfieldCheckInstance
    org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__DELETE( StatusfieldCheckInstance);
}

int Set_DSG_Mode1234::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // operationalMode
    {
        operationalMode=(istream.readUI8());
        retVal+=1;
    }
    // StatusfieldCheckInstance
    if ((operationalMode == 0x03) ||                (operationalMode ==0x04)) {
        org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__NEW( StatusfieldCheckInstance);
        StatusfieldCheckInstance->parent=this;
        retVal += StatusfieldCheckInstance->read(istream);
    }
    return retVal;
}

int Set_DSG_Mode1234::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read operationalMode
    {
        operationalMode=(istream.readUI8());
        retVal+=1;
    }
    // read StatusfieldCheckInstance
    if ((operationalMode == 0x03) ||                (operationalMode ==0x04)) {
        org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__NEW( StatusfieldCheckInstance);
        StatusfieldCheckInstance->parent=this;
        retVal += StatusfieldCheckInstance->read(istream);
    }

    return retVal;
}

int Set_DSG_Mode1234::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }


    {   /** fix dependent sizes for StatusfieldCheckInstance **/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write operationalMode
    ostream.writeUI8(operationalMode);
    retVal +=1;
    // write StatusfieldCheckInstance
    if ( StatusfieldCheckInstance != NULL ) {
        retVal +=StatusfieldCheckInstance->write(ostream);
    }

    return retVal;
}

int Set_DSG_Mode1234::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("Set_DSG_Mode1234\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write operationalMode
    dc.indent();
    dc.getPs().print("operationalMode=");
    dc.getPs().println((long)operationalMode);
    // write StatusfieldCheckInstance
    if ( StatusfieldCheckInstance != NULL ) {
        dc.indent();
        dc.getPs().println("StatusfieldCheckInstance") ;
        retVal +=StatusfieldCheckInstance->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE Set_DSG_Mode1234::getHeader()
//{
//return this->header;
//}

// Setter for header
//void Set_DSG_Mode1234::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 Set_DSG_Mode1234::getLength()
//{
//return this->length;
//}

// Setter for length
//void Set_DSG_Mode1234::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for operationalMode
//ui8 Set_DSG_Mode1234::getOperationalMode()
//{
//return this->operationalMode;
//}

// Setter for operationalMode
//void Set_DSG_Mode1234::setOperationalMode(ui8 val)
//{
//this->operationalMode= val;
//}

// Getter for StatusfieldCheckInstance
//org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__PTR_TYPE Set_DSG_Mode1234::getStatusfieldCheckInstance()
//{
//return this->StatusfieldCheckInstance;
//}

// Setter for StatusfieldCheckInstance
//void Set_DSG_Mode1234::setStatusfieldCheckInstance(org_himalay_ccif_Set_DSG_Mode1234_StatusfieldCheck__PTR_TYPE val)
//{
//this->StatusfieldCheckInstance= val;
//}

int Set_DSG_Mode1234::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void Set_DSG_Mode1234::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE Set_DSG_Mode1234::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_Set_DSG_Mode1234__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::Set_DSG_Mode1234::StatusfieldCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::Set_DSG_Mode1234* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::Set_DSG_Mode1234* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code